pypeit/core/fitting.py

```diff
@@ -759,7 +759,7 @@ def iterfit(xdata, ydata, invvar=None, inmask=None, upper=5, lower=5, x2=None,
     while (error != 0 or qdone is False) and iiter <= maxiter:
         goodbk = sset.mask.nonzero()[0]
         if maskwork.sum() <= 1 or not sset.mask.any():
-            sset.coeff = 0
+            sset.coeff[:] = 0
```

Comment:
Should this be 0. since it is a float. Not sure if that matters, but for consistency with above.

Reply:
This is a good point.  I did a quick check:
```
>>> import numpy
>>> x = numpy.arange(10).astype(float)
>>> x
array([0., 1., 2., 3., 4., 5., 6., 7., 8., 9.])
>>> x[:] = 0
>>> x
array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])
>>> x.dtype
dtype('float64')
```
So, I think it should be okay, but maybe better to be safe than sorry.